source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -484,26 +510,26 @@ def updateContextRect(self, context, rect=None, obj=None):
 		if rect is None:
 			try:
 				rect = getContextRect(context, obj=obj)
-			except (LookupError, NotImplementedError, RuntimeError, TypeError):
+			except (LookupError, RuntimeError, TypeError):
 				rect = None
 		self.contextToRectMap[context] = rect
 
-	def handleFocusChange(self, obj):
+	def handleFocusChange(self, obj: "NVDAObject"):
 		self.updateContextRect(context=Context.FOCUS, obj=obj)
 		if not api.isObjectInActiveTreeInterceptor(obj):
 			self.contextToRectMap.pop(Context.BROWSEMODE, None)
 		else:
 			self.handleBrowseModeMove()
 
-	def handleReviewMove(self, context):
+	def handleReviewMove(self, context: Context):
 		self.updateContextRect(context=Context.NAVIGATOR)
 
-	def handleBrowseModeMove(self, obj=None):
+	def handleBrowseModeMove(self, obj: "CursorManager | None" = None):
 		self.updateContextRect(context=Context.BROWSEMODE)
 
 	def refresh(self):
 		"""Refreshes the screen positions of the enabled highlights."""
-		if self._window and self._window.handle:
+		if self._window and self._window.handle and self.lastContextToRectMap != self.contextToRectMap:
```

Comment:
I was a bit puzzled by this, but then noticed that equality checks actually work nicely here. e.g.:
```
a={Context.CARET: RectLTRB(1,2,3,4)}
b={Context.CARET: RectLTRB(1,2,3,4)}
a==b
```